### ISSUE TITLE
inf/nan detection on HP-UX

### ISSUE THREAD

**Issue by karel-m** [rurban/Cpanel-JSON-XS#56]
I have experienced the following failure when trying to build Cpanel-JSON-XS-3.0210 on HP-UX
* HP-UX B.11.23 U 9000/800
* compiler: gcc version 4.6.1 (Target: hppa64-hp-hpux11.11)
* perl-5.22.0 (osname=hpux, osvers=11.11, archname=PA-RISC2.0-thread-multi-LP64)

The failure looks like this
```
Entering Cpanel-JSON-XS-3.0210
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.04_01)
Running Makefile.PL
Configuring Cpanel-JSON-XS-3.0210 ... Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Cpanel::JSON::XS
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have Pod::Usage 1.33 ... Yes (1.67)
Checking if you have Pod::Text 2.08 ... Yes (3.18)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.04_01)
Building and testing Cpanel-JSON-XS-3.0210 ... cp XS.pm blib/lib/Cpanel/JSON/XS.pm
cp XS/Boolean.pm blib/lib/Cpanel/JSON/XS/Boolean.pm
Running Mkbootstrap for Cpanel::JSON::XS ()
chmod 644 "XS.bs"
"/opt/perl64/bin/perl" "/opt/perl64/lib/5.22.0/ExtUtils/xsubpp"  -typemap "/opt/perl64/lib/5.22.0/ExtUtils/typemap" -typemap "typemap"  XS.xs > XS.xsc && mv XS.xsc XS.c
gcc64 -c   -D_POSIX_C_SOURCE=199506L -D_REENTRANT -mpa-risc-2-0 -fPIC -DPERL_DONT_CREATE_GVSV -D_HPUX_SOURCE -fwrapv -fno-strict-aliasing -pipe -I/usr/local/pa20_64/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2 -g -O   -DVERSION=\"3.0210\" -DXS_VERSION=\"3.0210\" -fPIC "-I/opt/perl64/lib/5.22.0/PA-RISC2.0-thread-multi-LP64/CORE"   XS.c
rm -f blib/arch/auto/Cpanel/JSON/XS/XS.sl
/usr/bin/ld  -b -L/usr/local/pa20_64/lib -L/usr/local/lib/pa20_64 -L/usr/lib/pa20_64 -L/usr/local/lib -L/pro/local/lib -L/lib/pa20_64 XS.o  -o blib/arch/auto/Cpanel/JSON/XS/XS.sl      \
                \

Warning: Some debug info sections were missing.
PXDB aborted.
chmod 755 blib/arch/auto/Cpanel/JSON/XS/XS.sl
"/opt/perl64/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- XS.bs blib/arch/auto/Cpanel/JSON/XS/XS.bs 644
cp bin/cpanel_json_xs blib/script/cpanel_json_xs
"/opt/perl64/bin/perl" -MExtUtils::MY -e 'MY->fixin(shift)' -- blib/script/cpanel_json_xs
Manifying 1 pod document
Manifying 2 pod documents
Running Mkbootstrap for Cpanel::JSON::XS ()
chmod 644 "XS.bs"
PERL_DL_NONLAZY=1 "/opt/perl64/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00_load.t ................. ok
t/01_utf8.t ................. ok
t/02_error.t ................ ok
t/03_types.t ................ ok
t/04_dwiw_encode.t .......... ok
t/05_dwiw_decode.t .......... ok
t/06_pc_pretty.t ............ ok
t/07_pc_esc.t ............... ok
t/08_pc_base.t .............. ok
t/08_pc_base_nv.t ........... ok
t/09_pc_extra_number.t ...... ok
t/104_sortby.t .............. ok
t/105_esc_slash.t ........... ok
t/106_allow_barekey.t ....... ok
t/107_allow_singlequote.t ... ok
t/108_decode.t .............. ok
t/109_encode.t .............. ok
t/10_pc_keysort.t ........... ok
t/110_bignum.t .............. ok
t/112_upgrade.t ............. ok
t/113_overloaded_eq.t ....... ok
t/114_decode_prefix.t ....... ok
t/115_tie_ixhash.t .......... ok
t/116_incr_parse_fixed.t .... ok
t/117_numbers.t ............. 1/19
#   Failed test 'inf -> null'
#   at t/117_numbers.t line 7.
#          got: '[++]'
#     expected: '[null]'

#   Failed test 'nan -> null'
#   at t/117_numbers.t line 8.
#          got: '[-?]'
#     expected: '[null]'

#   Failed test '-inf -> null'
#   at t/117_numbers.t line 9.
#          got: '[---]'
#     expected: '[null]'

#   Failed test '-nan -> null'
#   at t/117_numbers.t line 10.
#          got: '[?]'
#     expected: '[null]'

#   Failed test '-nan -> null'
#   at t/117_numbers.t line 11.
#          got: '[?]'
#     expected: '[null]'

#   Failed test 'inf -> "inf"'
#   at t/117_numbers.t line 25.
#          got: '[++]'
#     expected: '["inf"]'

#   Failed test '-inf -> "-inf"'
#   at t/117_numbers.t line 26.
#          got: '[---]'
#     expected: '["-inf"]'

#   Failed test 'nan -> "nan"'
#   at t/117_numbers.t line 29.
#                   '[-?]'
#     doesn't match '(?^:\[\"(-nan|nan)\"\])'

#   Failed test '-nan -> "-nan"'
#   at t/117_numbers.t line 30.
#                   '[?]'
#     doesn't match '(?^:\[\"(-nan|nan)\"\])'

#   Failed test '-nan -> "-nan"'
#   at t/117_numbers.t line 31.
#                   '[?]'
#     doesn't match '(?^:\[\"(-nan|nan)\"\])'

#   Failed test 'inf'
#   at t/117_numbers.t line 34.
#          got: '[++]'
#     expected: '[inf]'

#   Failed test '-inf'
#   at t/117_numbers.t line 35.
#          got: '[---]'
#     expected: '[-inf]'

#   Failed test 'nan'
#   at t/117_numbers.t line 36.
#                   '[-?]'
#     doesn't match '(?^:\[(-nan|nan)\])'

#   Failed test '-nan'
#   at t/117_numbers.t line 37.
#                   '[?]'
#     doesn't match '(?^:\[(-nan|nan)\])'

#   Failed test '-nan'
#   at t/117_numbers.t line 38.
#                   '[?]'
#     doesn't match '(?^:\[(-nan|nan)\])'
# Looks like you failed 15 tests of 19.
t/117_numbers.t ............. Dubious, test returned 15 (wstat 3840, 0xf00)
Failed 15/19 subtests
t/11_pc_expo.t .............. ok
t/12_blessed.t .............. ok
t/13_limit.t ................ ok
t/14_latin1.t ............... ok
t/15_prefix.t ............... ok
t/16_tied.t ................. ok
t/17_relaxed.t .............. ok
t/18_json_checker.t ......... ok
t/19_incr.t ................. ok
t/20_faihu.t ................ ok
t/20_unknown.t .............. Subroutine JSON::PP::Boolean::(0+ redefined at /opt/perl64/lib/5.22.0/overload.pm line 50.
Subroutine JSON::PP::Boolean::(-- redefined at /opt/perl64/lib/5.22.0/overload.pm line 50.
Subroutine JSON::PP::Boolean::(++ redefined at /opt/perl64/lib/5.22.0/overload.pm line 50.
t/20_unknown.t .............. ok
t/21_evans.t ................ ok
t/22_comment_at_eof.t ....... ok
t/23_array_ctx.t ............ ok
t/24_freeze_recursion.t ..... ok
t/25_boolean.t .............. ok
t/52_object.t ............... ok
t/53_readonly.t ............. ok
t/54_stringify.t ............ ok
t/55_modifiable.t ........... ok
t/96_interop.t .............. skipped: JSON::XS and JSON required for testing interop
t/96_mojo.t ................. skipped: Mojo::JSON required for testing interop
t/97_unshare_hek.t .......... ok
t/98_56only.t ............... ok
t/99_binary.t ............... ok
t/z_kwalitee.t .............. skipped: This test is only run for the module author
t/z_leaktrace.t ............. skipped: require Test::LeakTrace
t/z_meta.t .................. skipped: This test is only run for the module author
t/z_perl_minimum_version.t .. skipped: Author tests not required for installation
t/z_pod-coverage.t .......... skipped: This test is only run for the module author
t/z_pod-spell-mistakes.t .... skipped: This test is only run for the module author
t/z_pod-spelling.t .......... skipped: This test is only run for the module author
t/z_pod.t ................... ok
t/zero-mojibake.t ........... ok

Test Summary Report
-------------------
t/117_numbers.t           (Wstat: 3840 Tests: 19 Failed: 15)
  Failed tests:  1-15
  Non-zero exit status: 15
Files=59, Tests=1519, 15 wallclock secs ( 1.33 usr  0.53 sys +  9.93 cusr  2.50 csys = 14.29 CPU)
Result: FAIL
```
FYI: similar failure occurs also with HP C compiler

**Comment by rurban** [comment]
Thanks. That's a HPUX libm glitch 5.20 cannot not handle yet. The inf/nan libm probes were only added with 5.22. I'll have to todo these inf/nan test. 
Do you have a 5.22 perl on HPUX also? Would be interesting to know if 5.22 fixed those results.

**Comment by karenetheridge** [comment]
This was also raised in the JSON-MaybeXS queue under https://rt.cpan.org/Ticket/Display.html?id=110807.

**Comment by rurban** [comment]
Fixed with 421911486efdc303b1d79a6277b795cbaec30a4 for 3.0217_01
If still an issue please re-open
